apps/omg_watcher/lib/block_getter/core.ex

```diff
@@ -172,7 +175,8 @@ defmodule OMG.Watcher.BlockGetter.Core do
   """
   @spec get_eth_range_for_block_submitted_events(t(), non_neg_integer()) :: {pos_integer(), pos_integer()}
   def get_eth_range_for_block_submitted_events(%__MODULE__{synced_height: synced_height}, coordinator_height) do
-    {synced_height + 1, coordinator_height}
+    {:ok, finality_margin} = Application.fetch_env(:omg_api, :eth_submission_finality_margin)
```

Comment:
since this function's complexity grew somewhat:
 - don't `Application.fetch_env` here, do it in `init` and keep `finality_margin` as part of `Core's` state, or at least in the caller of this. `Core` is sidecause/sideeffect free (except for `Logger` calls if necessary)
 - add a simple test, that at least tells us that we are looking "into the past"
 - avoid negative result! -  that deserves a test too!